lte/gateway/c/session_manager/PipelinedClient.h

```diff
@@ -155,8 +155,12 @@ class PipelinedClient {
       const SessionState::SessionInfo info,
       std::function<void(Status status, UPFSessionContextState)> callback) = 0;
 
-  virtual uint32_t get_next_teid()    = 0;
-  virtual uint32_t get_current_teid() = 0;
+  virtual uint32_t get_next_teid()      = 0;
+  virtual uint32_t get_current_teid()   = 0;
+  virtual std::string get_upf_n3_addr() = 0;
+  virtual std::string get_upf_node_id() = 0;
+  virtual bool set_upf_node(
+      const std::string& node_id, const std::string& n3_addr) = 0;
```

Comment:
should not be part of pipelined client...

Reply:
same as above +  As per 29.244 section 5.8.2  UPF teid allocation shall be done either by SMF or UPF, however though SMF allocates teid, it prefix details s are supposed shared by UPF during the node level association setup message .  As the prefix are shared by UPF and it's association/linked to “pipelined client”, I kept the “get_teid” logic in Pipelind_client object.   However in our case we decided to allocate UPF teid globally for any session. But in long run it make sense to have this allocation logic in “Pipelined”, reasons below 2
	In future, we cud allocate by linking this to pipelined whichever is active ?
	All sessions belong to one UPF node, might have same prefix teid (though we support 1 UPF node for now)

Reply:
PipelinedClient is not a "state store", it is a communication object.

Reply:
Moved all these to SessionStateEnforcer.cpp